var/spack/repos/builtin/packages/py-ray/package.py

```diff
@@ -20,13 +35,15 @@ class PyRay(PythonPackage):
     depends_on('py-setuptools', type='build')
     depends_on('py-cython@0.29.14:', type='build')
     depends_on('py-wheel', type='build')
-    depends_on('py-aiohttp', type=('build', 'run'))
+    depends_on('py-aiohttp', type=('build', 'run'), when='@:1.0.1')
+    depends_on('py-aiohttp@3.7', type=('build', 'run'), when='@1.1.0:1.7.0')
+    depends_on('py-aiohttp@3.7:', type=('build', 'run'), when='@1.7.1:')
     depends_on('py-aioredis', type=('build', 'run'))
     depends_on('py-click@7.0:', type=('build', 'run'))
-    depends_on('py-colorama', type=('build', 'run'))
-    depends_on('py-colorful', type=('build', 'run'))
+    depends_on('py-colorama', type=('build', 'run'), when='@:1.5.2')
+    depends_on('py-colorful', type=('build', 'run'), when='@:1.5.2')
     depends_on('py-filelock', type=('build', 'run'))
-    depends_on('py-google', type=('build', 'run'))
+    depends_on('py-google', type=('build', 'run'), when='@:1.0.1')
     depends_on('py-gpustat', type=('build', 'run'))
```

Comment:
Need to specify version required, should be `@1:` (Spack doesn't sort beta releases properly so don't use `@1.0.0b1:`)